NuRadioReco/utilities/cr_flux.py

```diff
@@ -0,0 +1,261 @@
+        p1 = -3.28
+        p2 = -2.68
+        p3 = -4.84
+        E1 = 10 ** 18.69
+        E2 = 10 ** 19.81
+        c = 2.24e-30
+        c1 = c * (E1 / 1e18) ** p1
+        c2 = c1 * (E2 / E1) ** p2
+        spectrum = np.where(energy < E1,
+                            c * (energy / 1e18) ** p1,
+                            np.where(energy < E2,
+                                     c1 * (energy / E1) ** p2,
+                                     c2 * (energy / E2) ** p3))
+
+    else:
+        raise NotImplementedError(f"Type {type} is not supported")
+
+    spectrum = spectrum * (units.eV * units.km ** 2 * units.sr * units.year) ** -1
+
+    return spectrum
+
+
+def get_flux_per_energy_bin(log10e_min, log10e_max, type='data_auger_19'):
+    """
+    Returns an scipy integration of the measured data or the analytic spectrum over given interval.
+
+    Parameters
+    ---------
+    log10e_min: float
+        Input energy lower bound (in log10(E / eV)))
+    log10e_max: float
+        Input energy upper bound (in log10(E / eV)))
+    type: string
+        choose between data_auger_19, data_TA_21, auger_17, auger_19, TA_19
+
+    Returns
+    -------
+    scipy integration of data in NuRadio base units (which are 1/(eV m^2 sr ns))
+    """
+    if type in ['data_auger_19', 'data_TA_21']:
+        E, J, J_err_low, J_err_up = get_measured_data(type)
+        flux_interpolation = get_flux_interpolation(type)
+        integrated_flux = quad(flux_interpolation, log10e_min, log10e_max,
+                               limit=2 * E.shape[0], points=E)
+
+    elif type in ['auger_17', 'auger_19', 'TA_19']:
+        def flux(x):
+            """ Bring parametrized energy spectrum in right shape for quad() function """
+            return get_analytic_cr_flux(np.log10(np.array([x])), type)[0]
+
+        integrated_flux = quad(flux, 10 ** log10e_min, 10 ** log10e_max)
+    else:
+        raise NotImplementedError(f"Type {type} is not supported")
+
+    return integrated_flux[0]
+
+
+def cr_event_rate(log10e_min, log10e_max=21, zenith_min=10, zenith_max=80, a_eff=1, type="auger_19"):
+    """
+    Cosmic ray event rate in specified energy range assuming a detector with effective area
+    'A_eff' and zenith range. The detector projection and zenith band are taken into account.
+    The flux is calculated with the analytic spectrum.
+
+    Parameters
+    ---------
+    log10e_min: float
+        lower energy for energy range, in units log10(energy/eV)
+    log10e_max: float
+        upper energy for energy range, in units log10(energy/eV)
+    zenith_min: float
+        lower zenith for zenith range in deg
+    zenith_max: float
+        upper zenith for zenith range in deg
+    a_eff:
+        effective area of detector
+    type: string
+        choose between auger_17, auger_19 and TA_19
+    Returns
+    -------
+    eventrate
+    """
+    projected_area = 0.5 * (np.cos(zenith_min * units.deg) + np.cos(zenith_max * units.deg))
```

Comment:
not sure if this is a safe way. if you are calculating for a large range in zenith (the default) you are just taking the middle.... whereas if you calculate over many small bins and sum, you likely will not get the same result.

Reply:
or will you? But you need to convince me first ;)

Reply:
Thanks for spotting this, I simplified the function so that only the differential for an energy and a zenith is given.